packages/tupaia-web-server/src/routes/UserRoute.ts

```diff
@@ -28,8 +28,41 @@ export class UserRoute extends Route<UserRequest> {
       first_name: firstName,
       last_name: lastName,
       email,
+      preferences,
     } = await ctx.services.central.getUser();
 
-    return { userName: `${firstName} ${lastName}`, email };
+    const userResponse: {
+      userName: string;
+      email: string;
+      project?: {
+        id: string;
+        name: string;
+        code: string;
+        homeEntityCode: string;
+        dashboardGroupName: string;
+        defaultMeasure: string;
+      };
+    } = { userName: `${firstName} ${lastName}`, email };
+
+    if (preferences?.project_id) {
+      const { projects = [] } = await ctx.services.webConfig.fetchProjects({
+        showExcludedProjects: false,
+      });
+
+      const { id, name, code, homeEntityCode, dashboardGroupName, defaultMeasure } = projects.find(
```

Comment:
Since this is basically doing the same thing as the login route, I wonder if we can move the logic here to a util?